### PR TITLE
new nav flow

### PR DESCRIPTION
**Pull request by bencbernstein**

### CODE REVIEW

File: Track Your Reps/AppDelegate.swift [hunk 1]
```diff
@@ -9,26 +9,31 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
-        navigationController.navigationBar.isTranslucent = false
-        navigationController.navigationBar.barTintColor = Palette.pink.color
-        setupWindow(with: navigationController)
+        if userState == nil {
+            let onboardVC = OnBoardViewController()
+            setupWindow(with: onboardVC)
+        } else {
+            let tabBarController = TabBarController()
+            setupWindow(with: tabBarController)
+        }
```
Comment (OLI9292) [inline Track Your Reps/AppDelegate.swift:27]: could shorten to `initialView = userState == nil ? OnBoardViewController() : TabBarController(); setupWindow(with: initialView)`

File: Track Your Reps/TabBarController.swift [hunk 2]
```diff
@@ -10,29 +10,21 @@ protocol SetupViewControllersDelegate: class {
...
-    }
-    
-    func setupUser() {
-        guard let userState = userState else { onboardUser(); return }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
         User.sharedInstance.delegate = self
-        User.sharedInstance.state = userState
+        User.sharedInstance.state = userState!
```
Comment (OLI9292) [inline Track Your Reps/TabBarController.swift:21]: maybe something safer like `User.sharedInstance.state = userState ?? "NY"`

Comment (bencbernstein) [reply Track Your Reps/TabBarController.swift:21]: yeah will do

File: Track Your Reps/AppDelegate.swift (hunk 1 above)
Comment (bencbernstein) [reply Track Your Reps/AppDelegate.swift:27]: will do

File: Track Your Reps/Views/OnboardVC.swift [hunk 3]
```diff
@@ -15,17 +15,22 @@ class OnBoardViewController: UIViewController, UIPickerViewDelegate, UIPickerVie
...
         setupView()
     }
-
+    
     func buttonAction(sender: UIButton!) {
         let state = states[pickerView.selectedRow(inComponent: 0)]
         UserDefaults.standard.set(state, forKey: "state")
-        super.dismiss(animated: true, completion: nil)
+
+        present(TabBarController(), animated: true) {
```
Comment (OLI9292) [inline Track Your Reps/Views/OnboardVC.swift:28]: can just do `present(TabBarController(), animated: true, completion: nil)`

**Pull request merged**